psl-core/src/main/java/org/linqs/psl/util/RandUtils.java

```diff
@@ -43,13 +43,11 @@
     private RandUtils() {}
 
     private static synchronized void ensureRNG() {
-        if (rng != null) {
-            return;
+        if (rng == null) {
```

Comment:
Invert this guard (like we had it before).
As a rule of thumb, favor lower indentation levels.